### PR TITLE
feat(common): add .kpj typescript reader and schema 🙀

### PR DESCRIPTION
**Pull request by mcdurdin**
Part of #5283.

Baseline support for reading .kpj files in common/web/types. Groundwork for loading .kpj project files in kmc.

@keymanapp-test-bot skip

### CODE REVIEW

File: common/schemas/kpj/kpj.schema.json [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        "CheckFilenameConventions": {
+          "type": "string",
+          "pattern": "^(True|False)$"
+        },
+        "ProjectType": {
+          "type": "string",
+          "pattern": "^(keyboard|lexicalmodel)$"
+        },
+        "Version": {
+          "type": "string"
```
Comment (darcywong00) [inline common/schemas/kpj/kpj.schema.json:51]: Should version have a pattern involving \d+\.\d (plus optional engineering versions)?

Comment (mcdurdin) [reply common/schemas/kpj/kpj.schema.json:51]: Good idea!

File: common/schemas/kpj/kpj.schema.json [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+          "type": "string",
+          "pattern": "^(True|False)$"
+        },
+        "CheckFilenameConventions": {
+          "type": "string",
+          "pattern": "^(True|False)$"
+        },
+        "ProjectType": {
+          "type": "string",
+          "pattern": "^(keyboard|lexicalmodel)$"
```
Comment (darcywong00) [inline common/schemas/kpj/kpj.schema.json:48]: Is this `KeymanDeveloperProjectType` in keyman-developer-project.ts?
```suggestion
          "pattern": "^(Keyboard|LexicalModel)$"
```

Comment (mcdurdin) [reply common/schemas/kpj/kpj.schema.json:48]: No, that is an enum type in TypeScript, this is the XML and should be lower case.

File: common/web/types/src/kpj/kpj-file.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  Options?: KPJFileOptions;
+  Files?: KPJFileFiles;
+};
+
+export interface KPJFileOptions {
+  BuildPath?: string;
+  CompilerWarningsAsErrors?: string; // default False
+  WarnDeprecatedCode?: string; // default True
+  CheckFilenameConventions?: string; // default True
+  ProjectType?: 'keyboard' | 'lexicalmodel';
```
Comment (darcywong00) [inline common/web/types/src/kpj/kpj-file.ts:23]: Are these defaults for comments or TODOs?
And is default ProjectType keyboard?

File: common/schemas/kpj/kpj.schema.json (hunk 1 above)
Comment (mcdurdin) [reply common/schemas/kpj/kpj.schema.json:51]: Adding this in follow-up PR

File: common/web/types/src/kpj/kpj-file.ts (hunk 3 above)
Comment (mcdurdin) [reply common/web/types/src/kpj/kpj-file.ts:23]: These defaults are just comments

**Review by darcywong00 — APPROVED** [review]
lgtm

**Pull request merged**